deps/rabbit/src/rabbit_db_policy.erl

```diff
@@ -0,0 +1,69 @@
+%% This Source Code Form is subject to the terms of the Mozilla Public
+%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%
+%% Copyright (c) 2007-2023 VMware, Inc. or its affiliates.  All rights reserved.
+%%
+
+-module(rabbit_db_policy).
+
+-include_lib("rabbit_common/include/rabbit.hrl").
+-include("amqqueue.hrl").
+
+-export([update/3]).
+
+%% -------------------------------------------------------------------
+%% update().
+%% -------------------------------------------------------------------
+
+-spec update(VHostName, UpdateXFun, UpdateQFun) -> Ret when
+      VHostName :: vhost:name(),
+      Exchange :: rabbit_types:exchange(),
+      Queue :: amqqueue:amqqueue(),
+      UpdateXFun :: fun((Exchange) -> #{exchange => Exchange,
+                                        update_function => fun((Exchange) -> Exchange)}),
+      UpdateQFun :: fun((Queue) -> #{queue => Queue,
+                                     update_function => fun((Queue) -> Queue)}),
+      Ret :: {[{Exchange, Exchange}], [{Queue, Queue}]}.
+
+update(VHost, GetUpdatedExchangeFun, GetUpdatedQueueFun) ->
+    rabbit_db:run(
+      #{mnesia => fun() -> update_in_mnesia(VHost, GetUpdatedExchangeFun, GetUpdatedQueueFun) end
+       }).
+
+%% [1] We need to prevent this from becoming O(n^2) in a similar
+%% manner to rabbit_binding:remove_for_{source,destination}. So see
+%% the comment in rabbit_binding:lock_route_tables/0 for more rationale.
```

Comment:
This is not really related to this PR since this comment only moved but this `rabbit_binding:lock_route_tables/0` function looks like it hasn't existed for a while. I'm not quite sure which comment in `rabbit_binding` this is referring to now. Maybe this can be removed?

Reply:
I have no clue either, I left it just in case. Legacy code can be full of surprises. It will eventually be removed with mnesia.